# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/style/PostTraversalTask.h
# Commit: 095b3edec3c8
# Full Hash: 095b3edec3c8ba9318c71d0acc91e10ce57818dc
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2019-04-29 21:53:38
# Regressor Bug: 1514869
# File Overlap Count: 1
# Description:
#   Bug 1514869 - patch 2 - Adapt platform-font-list code to work with either the existing in-process font list or cross-process shared font list. r=jwatt
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D22938
# ==============================================================================

diff -r 21ef00977ab6 -r 095b3edec3c8 layout/style/PostTraversalTask.h
--- a/layout/style/PostTraversalTask.h	Sat Apr 27 15:37:29 2019 +0000
+++ b/layout/style/PostTraversalTask.h	Sat Apr 27 15:37:58 2019 +0000
@@ -12,10 +12,14 @@
 /* a task to be performed immediately after a Servo traversal */
 
 namespace mozilla {
+class ServoStyleSet;
 namespace dom {
 class FontFace;
 class FontFaceSet;
 }  // namespace dom
+namespace fontlist {
+struct Family;
+}  // namespace fontlist
 }  // namespace mozilla
 class gfxUserFontEntry;
 
@@ -68,6 +72,18 @@
     return task;
   }
 
+  static PostTraversalTask InitializeFamily(fontlist::Family* aFamily) {
+    auto task = PostTraversalTask(Type::InitializeFamily);
+    task.mTarget = aFamily;
+    return task;
+  }
+
+  static PostTraversalTask FontInfoUpdate(ServoStyleSet* aSet) {
+    auto task = PostTraversalTask(Type::FontInfoUpdate);
+    task.mTarget = aSet;
+    return task;
+  }
+
   void Run();
 
  private:
@@ -90,6 +106,12 @@
 
     // mTarget (gfxUserFontEntry*)
     LoadFontEntry,
+
+    // mTarget (fontlist::Family*)
+    InitializeFamily,
+
+    // mTarget (ServoStyleSet*)
+    FontInfoUpdate,
   };
 
   explicit PostTraversalTask(Type aType)